### PR TITLE
PIL-1267 Field validation for UKTR LiableEntity data

### PR DESCRIPTION
**Pull request by stipoa**
PIL-1267 covers field-validation for these 6 fields in "liableEntities": 
              { "ukChargeableEntityName",
                "idType",
                "idValue",
                "amountOwedDTT",
                "amountOwedIIR",
                "amountOwedUTPR"
              }
PIL-1266 covers the other fields in the Submit UKTR message :
{        "accountingPeriodFrom"
         "accountingPeriodTo"
         "obligationMTT"
         "electionUKGAAP"
         "liabilities": {
           "electionDTTSingleMember": false,
           "electionUTPRSingleMember": false,
           "numberSubGroupDTT": 4,
           "numberSubGroupUTPR": 5,
           "totalLiability": 10000.99,
           "totalLiabilityDTT": 5000.99,
           "totalLiabilityIIR": 4000,
           "totalLiabilityUTPR": 10000.99,
...
}

### CODE REVIEW

**Review by olakunlekuye2 — APPROVED** [review]

File: app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UktrSubmission.scala [hunk 1]
```diff
@@ -30,10 +33,27 @@ trait UktrSubmission {
 }
 
 object UktrSubmission {
+  val UKTR_STUB_PROCESSING_DATE = "2022-01-31T09:26:17Z"
```
Comment (MoDweik) [inline app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UktrSubmission.scala:36]: why is this date static?

Comment (stipoa) [reply app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UktrSubmission.scala:36]: This was done to follow the convention used elsewhere in the stubs: a statis date+time is put into the response, and this date+time is then cross-checked in the unit test. I can remove this, and populate the response with the actual date/time now, if you think that makes more sense.

File: app/uk/gov/hmrc/pillar2externalteststub/controllers/SubmitUKTRController.scala [hunk 2]
```diff
@@ -16,39 +16,75 @@
...
+        }
+    }
+
+  def validateUktrSubmissionData(req: UktrSubmissionData): Future[Either[NonEmptyChain[ValidationError], UktrSubmissionData]] = {
+    val validationResult: ValidationResult[UktrSubmissionData] = req.validate
+    Future.successful(validationResult.toEither)
+  }
+
+  def validateNilReturn(nilReturnRequest: UktrSubmissionNilReturn): Future[Either[NonEmptyChain[ValidationError], UktrSubmissionNilReturn]] =
+    // Placeholder for NilReturn validation
```
[the remark this answers is not in the document]
Comment (stipoa) [reply app/uk/gov/hmrc/pillar2externalteststub/controllers/SubmitUKTRController.scala]: resolved.

File: app/uk/gov/hmrc/pillar2externalteststub/models/uktr/error/UktrErrors.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  processingDate: String,
+  code:           String,
+  text:           String
+)
+
+object UktrBusinessValidationErrorDetail {
+  implicit val format: OFormat[UktrBusinessValidationErrorDetail] = Json.format[UktrBusinessValidationErrorDetail]
+}
+
+// Error response objects
```
[the remark this answers is not in the document]
Comment (stipoa) [reply app/uk/gov/hmrc/pillar2externalteststub/models/uktr/error/UktrErrors.scala]: resolved.

File: app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UktrSubmission.scala (hunk 1 above)
Comment (stipoa) [reply app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UktrSubmission.scala:36]: New commit added to set stub processing date/timezone to current date/timezone for LiableEntity tests.

**Review by kevincoatsworth — APPROVED** [review]

**Comment by stipoa** [comment]
QA are happy for this to be merged into main.

**Pull request merged**